lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystem.java

```diff
@@ -169,8 +169,8 @@ public void deleteFiles(Collection<Location> locations)
             for (List<Location> locationBatch : partition(locations, batchSize)) {
                 StorageBatch batch = storage.batch();
                 for (Location location : locationBatch) {
-                    getBlob(storage, new GcsLocation(location))
-                            .ifPresent(blob -> batch.delete(blob.getBlobId()));
+                    GcsLocation gcsLocation = new GcsLocation(location);
+                    batch.delete(BlobId.of(gcsLocation.bucket(), gcsLocation.path()));
```

Comment:
I assume this does not throw if the blob does not exist.

Reply:
It doesn't throw:

```
Calling StorageBatchResult.get() on the return value yields true upon successful deletion, false if the blob was not found, or throws a StorageException if the operation failed.
```